File: majority_element.js
var majority_element = function(nums) {
  const dict = new Map();
  let max = 0;
  let ans = null;

  nums.forEach(function(num) {
    if (!dict[num]) {
      dict[num] = 0;
    }
    dict[num] += 1;
  });

  Object.keys(dict).forEach(function(key) {
    if (dict[key] > max) {
      max = dict[key];
      ans = key;
    }
  });

  return ans;
}

File: majority_element.rb
# @param {Integer[]} nums
# @return {Integer}
def majority_element(nums)
  dict = {}
  max = 0

  nums.each do |num|
    dict[num] ||= 0
    dict[num] += 1
  end

  max = dict.values.max

  dict.key(max)
end
